Source/WebCore/dom/Element.cpp

```diff
@@ -4601,7 +4601,7 @@ const RenderStyle* Element::resolveComputedStyle(ResolveComputedStyleMode mode)
     for (auto& element : makeReversedRange(elementsRequiringComputedStyle)) {
         if (computedStyle && computedStyle->containerType() != ContainerType::Normal && mode != ResolveComputedStyleMode::Editability) {
             // If we find a query container we need to bail out and do full style update to resolve it.
-            if (document->updateStyleIfNeeded())
+            if (!document->inStyleRecalc() && document->updateStyleIfNeeded())
```

Comment:
I don't think this is the correct fix although this might be a valid hardening measure.
We need to re-order things so that `RenderTheme::adjustAppearanceForElement` doesn't depend on `Element::computedStyle`. As Antti noted in the radar, "[y]ou can’t call computedStyle() from middle of a style recalc. These sort of shadow style adjustments based on host style should be done in resolveCustomStyle which gets it as a parameter (or we should pass the host style to style adjuster)."